pygmt/src/text.py

```diff
@@ -164,24 +163,27 @@ def text_(
         for texts, but this option is only valid if using x/y/text.
     {w}
     """
-
-    # pylint: disable=too-many-locals
     # pylint: disable=too-many-branches
-
     kwargs = self._preprocess(**kwargs)  # pylint: disable=protected-access
 
     # Ensure inputs are either textfiles, x/y/text, or position/text
     if position is None:
+        if (x is not None or y is not None) and textfiles is not None:
+            raise GMTInvalidInput(
+                "Provide either position only, or x/y pairs, or textfiles."
+            )
         kind = data_kind(textfiles, x, y, text)
+        if kind == "vectors" and text is None:
+            raise GMTInvalidInput("Must provide text with x/y pairs")
     else:
-        if x is not None or y is not None:
+        if x is not None or y is not None or textfiles is not None:
             raise GMTInvalidInput(
-                "Provide either position only, or x/y pairs, not both"
+                "Provide either position only, or x/y pairs, or textfiles."
             )
-        kind = "vectors"
-
-    if kind == "vectors" and text is None:
-        raise GMTInvalidInput("Must provide text with x/y pairs or position")
+        if text is None or is_nonstr_iter(text):
+            raise GMTInvalidInput("Text can't be None or array.")
+        kind = ""
+        textfiles = ""
```

Comment:
Why are the `kind` and `textfiles` variables set to blank strings here?

Reply:
Ok with `textfiles = ""`, didn't know about the `+tSTRING` syntax!

Looking at the `kind` variable, it seems to be an internal PyGMT-only variable that is set to bypass the if-condition `if kind == "vectors":` at L225. Maybe use `kind=None` instead?

```suggestion
        kind = None
        textfiles = ""
```